### PR TITLE
putting service-catalog resources into helm charts

### PR DESCRIPTION
**Pull request by arschles**
These charts are useful only if you run the service-catalog API server in TPR storage mode for now. Later, when we run behind the aggregator, they'll be useful for both storage modes.

This patch does not remove the manifests in contrib/examples/apiserver, as these are used by the walkthrough and needed if talking directly to the service-catalog API server (e.g. etcd mode or TPR mode but still want to talk to the API server).

### CODE REVIEW

File: charts/resources/broker/templates/broker-auth-secret.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+kind: Secret
+metadata:
+  name: {{ .Values.auth.secretName }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+type: Opaque
+data:
```
Comment (pmorie) [inline charts/resources/broker/templates/broker-auth-secret.yaml]: you can use `stringData` and skip the base64 encoding.

File: charts/resources/instance/templates/instance.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+apiVersion: servicecatalog.k8s.io/v1alpha1
+kind: Instance
+metadata:
+  name: {{ .Values.name }}
+spec:
+  serviceClassName: {{ .Values.serviceClassName }}
+  planName: {{ .Values.planName }}
```
Comment (vaikas-google) [inline charts/resources/instance/templates/instance.yaml]: Could we add parameters here to see how they would line up? Right now this seems like a lot of overhead to create a chart and a helm install, when a simple kubectl create would suffice.

Comment (arschles) [inline charts/resources/instance/templates/instance.yaml]: @vaikas-google 
>Could we add parameters here to see how they would line up?

are you asking for a list of params in the README for this chart?

>Right now this seems like a lot of overhead to create a chart and a helm install, when a simple kubectl create would suffice.

It is overkill right now for sure. This PR is preparing for the following situation, which is coming in the near future:

- We are behind the aggregator
- A new user wants a quick way to add a broker, provision, or bind

This chart will provide an easy way to quickly do operations against our API.

Let me know what you think @vaikas-google

File: charts/resources/broker/templates/broker-auth-secret.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  username: {{ .Values.auth.username | b64enc }}
```
Comment (krancour) [inline charts/resources/broker/templates/broker-auth-secret.yaml]: Nit (applicable in a few places throughout): there's no need to _pipe_ the value into `b64enc`. The following would work just as well, and imo, reads much cleaner: `{{ b64enc .Values.auth.username }}`

File: charts/resources/broker/templates/broker-auth-secret.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  password: {{ .Values.auth.password | b64enc }}
```
Comment (krancour) [inline charts/resources/broker/templates/broker-auth-secret.yaml]: Nit: there's no need to _pipe_ the value into `b64enc`. The following would work just as well, and imo, reads much cleaner:

```
username: {{ b64enc .Values.auth.username }}
password: {{ b64enc .Values.auth.password }}
```

File: charts/resources/instance/templates/instance.yaml (hunk 2 above)
Comment (krancour) [inline charts/resources/instance/templates/instance.yaml]: I have to echo @vaikas-google's questions. Even though you have addressed them already, @arschles, I don't feel I understand.

Let me say that I _do_ see the value and the power behind some future demo wherein, for instance, an instance and a binding could _both_ be included in a single chart and service catalog will sort everything out and the whole system will be eventually consistent. (To make this even more concrete, imagine augmenting a chart like the existing Drupal or Wordpress charts to use instance and binding resources to express a dependency on an rdbms without much regard for _how_ it's provisioned.)

But if these new charts, as PR'ed here, just contain a single resource each, aren't they not doing anything more than what the example manifests already do?

File: charts/resources/broker/templates/broker-auth-secret.yaml (hunk 1 above)
Comment (krancour) [inline charts/resources/broker/templates/broker-auth-secret.yaml]: Mind blown. Didn't know that was an option even. It's not documented, although I have found a scant few examples of it in the wild. Learned something new.

**Pull request closed without merging**